### PR TITLE
update on #9 to fit new project design and resolve conflicts

### PR DESCRIPTION
**Pull request by noamsabb**

### CODE REVIEW

File: src/server/java/server/Player.java [hunk 1]
```diff
@@ -32,34 +36,66 @@ public void replaceTiles(Tile[] currentTiles, Tile[] newTiles) {
...
+            sendToPlayer.accept(boardString);
+
+            System.out.println("Bag sent to the player");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
 
     public void sendPlayers(Player[] players) {
         throw new UnsupportedOperationException();
```
Comment (edenfar) [inline src/server/java/server/Player.java:86]: Implement this (it should send the data about all the players to the player)

File: src/server/java/server/Tile.java [hunk 2]
```diff
@@ -31,6 +32,17 @@ public boolean equals(Object obj) {
         return letter == other.letter && score == other.score;
     }
 
+    public static Tile fromString(String tileString) {
```
Comment (edenfar) [inline src/server/java/server/Tile.java:35]: Check if needed, new tiles should not be created

File: src/server/java/server/Player.java [hunk 3]
```diff
@@ -32,34 +36,66 @@ public void replaceTiles(Tile[] currentTiles, Tile[] newTiles) {
...
     }
 
     public void sendCurrentPlayer(String name) {
-        throw new UnsupportedOperationException();
+        sendToPlayer.accept(name);
```
Comment (edenfar) [inline src/server/java/server/Player.java:90]: Add some prefix that will help the receive identify that it is a player name

**Review by edenfar — CHANGES_REQUESTED** [review]

File: src/client/java/model/Model.java [hunk 4]
```diff
@@ -36,7 +42,42 @@ private void sendMessage(String message) {
...
+
+                    System.out.println("Board received and deserialized");
+                } catch (IOException | ClassNotFoundException e) {
+                    e.printStackTrace();
+                }
+            }
+            if (Bag) {
+                try {
+                    // Convert the received board string back to a byte array
+                    byte[] bagBytes = response.getBytes();
```
[the remark this answers is not in the document]
Comment (noamsabb) [reply src/client/java/model/Model.java:65]: not sure it's really necessary

**Review by edenfar — CHANGES_REQUESTED** [review]

File: src/server/java/server/Player.java [hunk 5]
```diff
@@ -74,26 +81,66 @@ public void notifyIllegalWord(Word word) {
...
+            System.out.println("Bag sent to the player");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public ByteArrayOutputStream seriallizeObject(Object o) throws IOException {
+        ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
+        ObjectOutputStream objectOut = new ObjectOutputStream(byteOut);
+        objectOut.writeObject(o);
```
[the remark this answers is not in the document]
Comment (noamsabb) [reply src/server/java/server/Player.java:112]: We use it to write Java objects to an output stream
objectOut.writeObject(o); writes the serialized form of the object o to the objectOut stream, which in turn writes it to the byteOut stream

**Review by edenfar — APPROVED** [review]

**Pull request merged**